core/commands/add.go

```diff
@@ -120,7 +136,12 @@ remains to be implemented.
 					return
 				}
 
-				rootnd, err := addFile(n, file, outChan, progress, wrap, trickle)
+				// If the file is not a folder, then let's get the root of that
+				// folder and attempt to load the appropriate .ipfsignore.
+				localIgnorePatterns := checkForParentIgnorePatterns(file.FileName(), ignoreFilePatterns)
```

Comment:
the problem is that -- doing this here, in the `Run` function, means it's getting executed in the _daemon_ with the daemon chdir. 

we need to do all this part (the ignores) before hand, in the PreRun (or in the cmdslib? maybe the cmdslib needs an option to be given a function like `os.Walk` that decides whether to add files or not -- that way we can implement support for `.ipfsignore` outside the cmdslib)